templates/types/streaming/nextjs/app/components/ui/chat/chat-input.tsx

```diff
@@ -43,25 +74,62 @@ export default function ChatInput(
     setImageUrl(base64);
   };
 
+  const handleUploadCsvFile = async (file: File) => {
+    const content = await new Promise<string>((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsText(file);
+      reader.onload = () => resolve(reader.result as string);
+      reader.onerror = (error) => reject(error);
+    });
+    setUploadedCsv({
+      content,
+      filename: file.name,
+      filesize: file.size,
+    });
+  };
+
   const handleUploadFile = async (file: File) => {
     try {
-      if (props.multiModal && file.type.startsWith("image/")) {
+      if (file.type.startsWith("image/")) {
         return await handleUploadImageFile(file);
       }
+      if (file.type === "text/csv") {
+        return await handleUploadCsvFile(file);
+      }
       props.onFileUpload?.(file);
```

Comment:
How about other file types (pdf, json, txt,...)? i think we can show a popup error that file type is not supported and list the supported one.